tests/test_raft.rs

```diff
@@ -691,12 +823,62 @@ fn test_commit_without_new_term_entry() {
 }
 
 #[test]
+fn test_dueling_pre_candidates() {
+    let mut a = new_test_raft(1, vec![1, 2, 3], 10, 1, new_storage());
+    let mut b = new_test_raft(2, vec![1, 2, 3], 10, 1, new_storage());
+    let mut c = new_test_raft(3, vec![1, 2, 3], 10, 1, new_storage());
+    a.pre_vote = true;
+    b.pre_vote = true;
+    c.pre_vote = true;
+
+    let mut nt = Network::new(vec![Some(a), Some(b), Some(c)], true);
+    nt.cut(1, 3);
+
+    nt.send(vec![new_message(1, 1, MessageType::MsgHup, 0)]);
+    nt.send(vec![new_message(3, 3, MessageType::MsgHup, 0)]);
+    // 1 becomes leader since it receives votes from 1 and 2
```

Comment:
A blank line here.
Notice all the blank lines for code block in this test case.